launcher/ui/pages/instance/ExternalResourcesPage.cpp

```diff
@@ -157,8 +157,49 @@ void ExternalResourcesPage::removeItem()
 {
     if (!m_controlsEnabled)
         return;
-    
+
     auto selection = m_filterModel->mapSelectionToSource(ui->treeView->selectionModel()->selection());
+
+    int count = 0;
+    bool folder = false;
+    for (auto i : selection.indexes()) {
+        if (i.column() == 0) {
+            count++;
+
+            // if a folder is selected, show the confirmation dialog
+            if (m_model->at(i.row()).fileinfo().isDir())
+                folder = true;
+        }
+    }
+
+    bool enough = count > 1;
+
+    if (enough || folder) {
+        QString text;
+        if (enough)
+            text = tr("About to remove: %1 items\n"
+                      "This may be permanent and they will be gone from the folder.\n\n"
+                      "Are you sure?")
+                       .arg(count);
+        else
+            text = tr("About to remove: %1 (folder)\n"
+                      "This may be permanent and it will be gone from the parent folder.\n\n"
+                      "Are you sure?")
+                       .arg(m_model->at(selection.indexes().at(0).row()).fileinfo().fileName());
+
+        auto response = CustomMessageBox::selectable(this, tr("CAREFUL!"), text, QMessageBox::Warning, QMessageBox::Yes | QMessageBox::No,
+                                                     QMessageBox::No)
+                            ->exec();
+
+        if (response != QMessageBox::Yes)
+            return;
+    }
```

Comment:
```suggestion
    int resource_count = 0;
    bool is_folder = false;

    for (auto&& i : selection.indexes()) {
        if (i.column() != 0)
            continue;

        resource_count++;

        // if a folder is selected, show the confirmation dialog
        if (m_model->at(i.row()).type() == ResourceType::FOLDER)
            is_folder = true;
    }

    bool multiple_mods_being_deleted = resource_count > 1;

    QString text;

    if (multiple_mods_being_deleted) {
        text = tr("About to remove: %1 items\n"
                  "This may be permanent and they will be gone from the folder.\n\n"
                  "Are you sure?")
                   .arg(resource_count);
    } else if (is_folder) {
        text = tr("About to remove: %1 (folder)\n"
                  "This may be permanent and it will be gone from the parent folder.\n\n"
                  "Are you sure?")
                   .arg(m_model->at(selection.indexes().at(0).row()).fileinfo().fileName());
    }

    if (!text.isEmpty()) {
        auto response = CustomMessageBox::selectable(this, tr("CAREFUL!"), text, QMessageBox::Warning, QMessageBox::Yes | QMessageBox::No,
                                                     QMessageBox::No)
                            ->exec();

        if (response != QMessageBox::Yes)
            return;
    }
```

wouldn't it make sense to also have the confirmation when deleting a single mod? it's still a destructive action after all (maybe not as much as a lot of them at once, but still).

Reply:
I thought about this a lot, but I realised it would hurt UX for people deleting mods one-by-one. Most people will have another copy in their downloads folder, and if it's personal resource pack it will probably be a folder. On most systems it will go to trash, but a lot of items in trash will be annoying to restore especially without being to use Ctrl+Z like you usually would be in your file browser.

Maybe the warning could just be enabled for one item on Flatpak? I could also disable warnings for deleting screenshots (but not worlds, since they are whole folders) unless Flatpak is being used. Items can be permanently removed on other environments, but only if trashing fails.

Reply:
It'd be fine having a warning only on systems without trashing yeah, but note that it's not only Flatpak, other systems, even on newer qt versions, can also lack trash support, and in such cases the warning should still show up IMO. Maybe the best alternative for that would be to create a temp file and attempt to trash it to see if it succeeds on startup or something, to set a boolean on whether to warn or not in such cases.

Reply:
I think I'll keep it as-is for now. Removing a single mod is unlikely to cause much damage.